src/coreclr/vm/jithelpers.cpp

```diff
@@ -5085,33 +5085,69 @@ void JIT_Patchpoint(int* counter, int ilOffset)
 #endif
 
     // Find context for the original method
+    CONTEXT *pFrameContext = NULL;
+#if defined(TARGET_WINDOWS) && defined(TARGET_AMD64)
+    DWORD contextSize = 0;
+    ULONG64 xStateCompactionMask = 0;
+    DWORD contextFlags = CONTEXT_FULL;
+    if (Thread::AreCetShadowStacksEnabled())
+    {
+        xStateCompactionMask = XSTATE_MASK_CET_U;
+        contextFlags |= CONTEXT_XSTATE;
+    }
+
+    // The initialize call should fail but return contextSize
+    BOOL success = g_pfnInitializeContext2 ?
+        g_pfnInitializeContext2(NULL, contextFlags, NULL, &contextSize, xStateCompactionMask) :
+        InitializeContext(NULL, contextFlags, NULL, &contextSize);
+
+    _ASSERTE(!success && (GetLastError() == ERROR_INSUFFICIENT_BUFFER));
+
+    PVOID pBuffer = _alloca(contextSize);
```

Comment:
What is the context size we usually end up with here on current hw? I am wondering whether it would be better to allocate this on heap.

Reply:
If it only adds [XSAVE_CET_U_FORMAT feature](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/ns-wdm-_xsave_cet_u_format) , then this is probably not large.

Reply:
The contextSize is 0x4ff.